servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/NettyIoExecutor.java

```diff
@@ -33,4 +34,11 @@
      * @return an {@link Executor} which will use an {@link IoExecutor} thread for execution.
      */
     Executor asExecutor();
+
+    /**
+     * Determine if threads used by this {@link IoExecutor} are marked with {@link IoThreadFactory.IoThread} interface.
+     *
+     * @return {@code true} if supported
+     */
+    boolean isIoThreadSupported();
```

Comment:
Don't have a strong preference, but since we introduced this flag specifically for EventLoop use-cases, consider moving it to `EventLoopAwareNettyIoExecutor` interface.